app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -16,7 +17,11 @@ def create
       result = OtpVerificationForm.new(current_user, sanitized_otp_code).submit
       post_analytics(result)
       if result.success?
-        handle_valid_otp
+        next_url = nil
+        if UserSessionContext.confirmation_context?(context)
+          next_url = user_next_authentication_setup_path!
+        end
```

Comment:
Minor style remark: Rubocop may get upset depending on line length, but we may be able to have this on a single line:

```suggestion
        next_url = user_next_authentication_setup_path if UserSessionContext.confirmation_context?(context)
```

Reply:
Yea rubocop was not happy with the length. it preferred the if statement wrapper.